docs/architecture.md

```diff
@@ -0,0 +1,188 @@
+attach their application's routes to this Gateway if their application's hostname matches `*.example.com`.
+* *Application A* with two Pods deployed in the *applications* Namespace by *Application Developer A*. To expose the
+application to its clients (*Clients A*) via the host `a.example.com`, *Application Developer A* creates *HTTPRoute A*
+and attaches it to `Gateway AB`.
+* *Application B* with one Pod deployed in the *applications* Namespace by *Application Developer B*. To expose the
+application to its clients (*Clients B*) via the host `b.example.com`, *Application Developer B* creates *HTTPRoute B*
+and attaches it to `Gateway AB`.
+* *Public Endpoint*, which fronts the *NKG* Pod. This is typically a TCP load balancer (cloud, software, or hardware)
+or a combination of such load balancer with a NodePort Service. *Clients A* and *B* connect to their applications via
+the *Public Endpoint*.
+
+The connections related to client traffic are depicted by the yellow and purple arrows, while the black arrows represent
+access to the Kubernetes API. The resources within the cluster are color-coded based on the user responsible for their
+creation. For example, the Cluster Operator is denoted by the color green, indicating that they have created and manage
+all the green resources.
+
+> Note: For simplicity, many necessary Kubernetes resources like Deployment and Services aren't shown,
+> which the Cluster Operator and the Application Developers also need to create.
+
+Next, let's explore the NKG Pod.
+
+## The NGINX Kubernetes Gateway Pod
+
+The NGINX Kubernetes Gateway consists of three containers:
+
+1. `nginx`: the data plane. Consists of an NGINX master process and NGINX worker processes. The master process controls
+the worker processes. The worker processes handle the client traffic and load balance the traffic to the backend
+applications.
+2. `nginx-gateway`: the control plane. Watches Kubernetes objects and configures NGINX.
+3. `busybox`: initializes the NGINX config environment.
+
+These containers are deployed in a single Pod as a Kubernetes Deployment. The init container, `busybox`, runs before the
+`nginx` and `nginx-gateway` containers and creates directories and sets permissions for the NGINX process.
+
+The `nginx-gateway`, or the control plane, is a [Kubernetes controller][controller], written with
+the [controller-runtime][runtime] library. It watches Kubernetes objects (Services, Endpoints, Secrets, and Gateway API
+CRDs), translates them to nginx configuration, and configures NGINX. This configuration happens in two stages. First,
+NGINX configuration files are written to the NGINX configuration volume shared by the `nginx-gateway` and `nginx`
+containers. Next, the control plane reloads the NGINX process. This is possible because the two
+containers [share a process namespace][share], which gives the `nginx-gateway` process access to the `nginx` process.
+
```

Comment:
I would add a note:

Note: Because of the  process namespace sharing, containers have access to the other containers filesystem via the `/proc/$pid/root` link.

Reply:
I don't think adding that comment is necessary because we don't use that feature of shared process namespace.